File: problems/647/solution.cpp
class Solution {
public:
    int countSubstrings(string s) {
        int ans = 0;
        
        for (int i = 0; i < s.size(); i++) {
            ans += countPalindromicSubstrings(s, i, i);
            ans += countPalindromicSubstrings(s, i, i + 1);
        }
        
        return ans;
    }
    
    int countPalindromicSubstrings(const string& s, int lo, int hi) {
        int ans = 0;
        
        while (lo >= 0 and hi < s.size()) {
            if (s[lo] != s[hi]) {
                break;
            }
            
            lo--;
            hi++;
            ans++;
        }
        
        return ans;
    }
};

File: problems/647/solution.go
package main
func countSubstrings(s string) int {
    ans := 0
    
    for i := 0; i < len(s); i++ {
        ans += countPlaindromicSubstrings(s, i, i)
        ans += countPlaindromicSubstrings(s, i, i + 1)
    }
    
    return ans
}

func countPlaindromicSubstrings(s string, lo, hi int) int {
    ans := 0
    
    for lo >= 0 && hi < len(s) {
        if s[lo] != s[hi] {
            break
        }
        
        lo--
        hi++
        ans++
    }
    
    return ans
}